drivers/modem/hl7800.c

```diff
@@ -0,0 +1,4887 @@
+{
+	switch (state) {
+		PREFIXED_SWITCH_CASE_RETURN_STRING(HL7800, NOT_REGISTERED);
+		PREFIXED_SWITCH_CASE_RETURN_STRING(HL7800, HOME_NETWORK);
+		PREFIXED_SWITCH_CASE_RETURN_STRING(HL7800, SEARCHING);
+		PREFIXED_SWITCH_CASE_RETURN_STRING(HL7800, REGISTRATION_DENIED);
+		PREFIXED_SWITCH_CASE_RETURN_STRING(HL7800, OUT_OF_COVERAGE);
+		PREFIXED_SWITCH_CASE_RETURN_STRING(HL7800, ROAMING);
+		PREFIXED_SWITCH_CASE_RETURN_STRING(HL7800, EMERGENCY);
+		PREFIXED_SWITCH_CASE_RETURN_STRING(HL7800, UNABLE_TO_CONFIGURE);
+	default:
+		return "UNKNOWN";
+	}
+}
+
+static void set_network_state(enum mdm_hl7800_network_state state)
+{
+	ictx.network_state = state;
+	generate_network_state_event();
+}
+
+static void generate_network_state_event(void)
+{
+	struct mdm_hl7800_compound_event event;
+	event.code = ictx.network_state;
+	event.string = get_network_state_string(ictx.network_state);
+	LOG_INF("Network State: %d %s", ictx.network_state, event.string);
+	event_handler(HL7800_EVENT_NETWORK_STATE_CHANGE, &event);
+}
+
+/* Handler: +CEREG: <n>,<stat>[,[<lac>],[<ci>],[<AcT>]
+ *  [,[<cause_type>],[<reject_cause>] [,[<Active-Time>],[<Periodic-TAU>]]]]
+ */
+static bool on_cmd_network_report_query(struct net_buf **buf, uint16_t len)
+{
+	size_t out_len;
+	char value[MDM_MAX_RESP_SIZE];
+
+	out_len = net_buf_linearize(value, sizeof(value), *buf, 0, len);
+	char *pos = strchr(value, ',');
+	if (pos) {
+		int l = (value + out_len) - pos;
+		char val[l];
+		strncpy(val, pos + 1, l);
+		val[l] = 0;
+		set_network_state(strtol(val, NULL, 0));
+
+		/* start work to adjust iface */
+		k_delayed_work_cancel(&ictx.iface_status_work);
+		k_delayed_work_submit_to_queue(&hl7800_workq,
+					       &ictx.iface_status_work,
+					       IFACE_WORK_DELAY);
+	}
+
+	return true;
+}
+
+#ifdef CONFIG_NEWLIB_LIBC
+/* Handler: +CCLK: "yy/MM/dd,hh:mm:ss±zz" */
+static bool on_cmd_rtc_query(struct net_buf **buf, uint16_t len)
+{
+	struct net_buf *frag = NULL;
+	size_t str_len = strlen(TIME_STRING_FORMAT);
+	char rtc_string[sizeof(TIME_STRING_FORMAT)];
+	memset(rtc_string, 0, sizeof(rtc_string));
+	ictx.local_time_valid = false;
+
+	wait_for_modem_data_and_newline(buf, net_buf_frags_len(*buf),
+					sizeof(TIME_STRING_FORMAT));
+
+	frag = NULL;
+	len = net_buf_findcrlf(*buf, &frag);
+	if (!frag) {
+		goto done;
+	}
+	if (len != str_len) {
+		LOG_WRN("Unexpected length for RTC string %d (expected:%d)",
+			len, str_len);
+	} else {
+		net_buf_linearize(rtc_string, str_len, *buf, 0, str_len);
+		LOG_INF("RTC string: '%s'", log_strdup(rtc_string));
+		ictx.local_time_valid = convert_time_string_to_struct(
+			&ictx.local_time, &ictx.local_time_offset, rtc_string);
+	}
+done:
+	return true;
+}
+
+static bool valid_time_string(const char *time_string)
+{
+	/* Ensure the all the expected delimiters are present */
+	size_t offset = TIME_STRING_DIGIT_STRLEN + TIME_STRING_SEPARATOR_STRLEN;
+	size_t i = TIME_STRING_FIRST_SEPARATOR_INDEX;
+	for (; i < TIME_STRING_PLUS_MINUS_INDEX; i += offset) {
```

Comment:
It is hard to review the kind of code where the variables and code are not separated. It would be helpful for the reviewer if there would be empty line between variables and code. Same applies to other functions in this file.